### ISSUE TITLE
Support qualified keys in maps

### ISSUE THREAD

**Issue by ikitommi** [metosin/malli#14]
Spec showed the way to reuse map key specs. This is kinda nice and would be easy to support: one could use explicitly registered leaf specs as maps key-value tuples, so:

```clj
(m/schema
  [:map
   [:id int?]
   [:name string?]])
```

could be written as:

```clj
(def registry
  (merge m/default-registry {::id int?}))

(m/schema
  [:map
   ::id
   [:name string?]]
  {:registry registry})
```

**Comment by ikitommi** [comment]
closed, via #194, merged via #239 .

```clj
(m/validate
  [:map {:registry {::id int?
                    ::country string?}}
   ::id
   [:name string?]
   [::country {:optional true}]]
  {::id 1
   :name "kikka"})
; => true
```

**Comment by ikitommi** [comment]
<img width="623" alt="Screenshot 2020-08-06 at 17 38 49" src="https://user-images.githubusercontent.com/567532/89546483-55919480-d80d-11ea-80f7-70b843e94051.png">
